tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,39 @@
+"""Task 2 of Module 4."""
+from pathlib import Path
+import os
+from tkinter.filedialog import askdirectory
+
+
+def filling_in_the_gaps():
+    """
+    Find all files with a given prefix,
+    in a single folder and fill in any gaps in the numbering.
+    """
+    dir_path = askdirectory(title='Select Folder with spam* files')
+    digit = 1
+    dozen = 0
+    hundred = 0
+
+    os.chdir(dir_path)
+
+    for file in os.listdir(dir_path):
+        file_path = Path(file)
+        files_number = file_path.stem[4:7]
+        expected_numbers = str(hundred) + str(dozen) + str(digit)
```

Comment:
It is an interesting algorithm, but Python can do it all for you :) All you need is a valid number of file and [str.zfill method](https://docs.python.org/3/library/stdtypes.html#str.zfill) or f-string `f'{file_number:03}'`.

Reply:
Algoritm replaced with introducing new variable: `expected_number = str(iter).zfill(3)`